### DISCUSSION TITLE
Image Variants in Cards

### DISCUSSION THREAD

**Discussion by cmaxw** [avo-hq/avo#3225] [Help] [answered]
Hey guys, whenever I try to add a variant to any of the ActiveStorage attached images to my cards, I get an error. 

I can load the original image, but I'd like to optimize my images a bit.

Here's an example of the image tag:

```ruby
            <%= image_tag meetup.meetup_series.thumbnail.variant(resize_to_limit: [320, 180]), class: "aspect-[16/9] w-full rounded-2xl bg-gray-100 object-cover sm:aspect-[2/1] lg:aspect-[3/2]" %>
```

I get this error: `13:15:36 web.1     | ArgumentError - Can't resolve image into URL: undefined method `to_model' for an instance of ActiveStorage::VariantWithRecord:`

If I call `#url` after the variant call, I get this error: `13:21:00 web.1     | ArgumentError - Nil location provided. Can't build URI.:`

I think this is a bug, but if I'm missing something, I'd appreciate any help.

**Comment by cmaxw** [comment, marked as answer]
Ok, I found this somewhere else in my code and it works:

```ruby
<%= image_tag main_app.url_for(meetup.meetup_series.thumbnail.variant(resize_to_limit: [320, 180])), class: "aspect-[16/9] w-full rounded-2xl bg-gray-100 object-cover sm:aspect-[2/1] lg:aspect-[3/2]" %>
```

**Comment by cmaxw** [comment]
I'm not sure if there's a "right" place to document this...

**Reply by Paul-Bob** [reply]
The right place is the docs sections about where you are trying to use them (resource tool, custom field or custom tool, etc...)
